cmd/insertvariantobject/insertvariantobject.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"encoding/json"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	cfg, err := sf.GetConfigFromEnv([]*sf.ConfigParam{
+		{Name: "Account", EnvName: "SNOWFLAKE_TEST_ACCOUNT", FailOnMissing: true},
+		{Name: "User", EnvName: "SNOWFLAKE_TEST_USER", FailOnMissing: true},
+		{Name: "Password", EnvName: "SNOWFLAKE_TEST_PASSWORD", FailOnMissing: true},
+		{Name: "Warehouse", EnvName: "SNOWFLAKE_TEST_WAREHOUSE", FailOnMissing: true},
+		{Name: "Database", EnvName: "SNOWFLAKE_TEST_DATABASE", FailOnMissing: true},
+		{Name: "Schema", EnvName: "SNOWFLAKE_TEST_SCHEMA", FailOnMissing: true},
+		{Name: "Host", EnvName: "SNOWFLAKE_TEST_HOST", FailOnMissing: false},
+		{Name: "Port", EnvName: "SNOWFLAKE_TEST_PORT", FailOnMissing: false},
+		{Name: "Protocol", EnvName: "SNOWFLAKE_TEST_PROTOCOL", FailOnMissing: false},
+	})
+	if err != nil {
+		log.Fatalf("failed to create Config, err: %v", err)
+	}
+	dsn, err := sf.DSN(cfg)
+	if err != nil {
+		log.Fatalf("failed to create DSN from Config: %v, err: %v", cfg, err)
+	}
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		log.Fatalf("failed to connect. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	param := map[string]string{"key": "value"}
+	jsonStr, err := json.Marshal(param)
+	if err != nil {
+		log.Fatalf("failed to marshal json. err: %v", err)
+	}
+
+	createTableQuery := "CREATE OR REPLACE TABLE insert_variant_object (c1 VARIANT, c2 OBJECT)"
+
+	// https://docs.snowflake.com/en/sql-reference/functions/parse_json
+	// can do with TO_VARIANT(PARSE_JSON(..)) as well, but PARSE_JSON already produces VARIANT
+	insertQuery := "INSERT INTO insert_variant_object (c1, c2) SELECT PARSE_JSON(?), TO_OBJECT(PARSE_JSON(?))"
+	// https://docs.snowflake.com/en/sql-reference/data-types-semistructured#object
+	insertOnlyObject := "INSERT INTO insert_variant_object (c2) SELECT OBJECT_CONSTRUCT('name', 'Jones'::VARIANT, 'age',  42::VARIANT)"
+
+	selectQuery := "SELECT c1, c2 FROM insert_variant_object"
+
+	fmt.Printf("Running CREATE OR REPLACE TABLE: %v\n", createTableQuery)
```

Comment:
Maybe just `Creating table: `?

And maybe it would be good to defer dropping table?

Reply:
I'm also thinking, maybe just to be on the safe side - it would be better to skip `OR REPLACE` clause, if we defer dropping. Let's consider the scenarios:
1. When table does not exist - it is created and dropped by us.
2. When table exists - it is better to fail on table creation, because we don't want to replace table, that is not managed by us.
I know chances are very low, but...

Reply:
you're right - although i do not expect anyone to actually run this code, but you got a very valid point
so i chose a different approach, suffixing the table name with `UnixNano` , hopefully reduces our chances very much :)  into having (and then, potentially dropping) a table with the exact name in the exact same database and exact same schema ..

also implemented the cleanup (drop table) which i also missed

Reply:
eventually, the drop has been moved to a function and that invoked from `defer`